src/shogun/evaluation/SigmoidCalibration.cpp

```diff
@@ -0,0 +1,188 @@
+	SG_ADD(&m_maxiter, "m_maxiter", "Maximum number of iteration for search.", MS_NOT_AVAILABLE);
+	SG_ADD(&m_minstep, "m_minstep", "Minimum step taken in line search.", MS_NOT_AVAILABLE);
+	SG_ADD(&m_sigma, "m_sigma", "Positive parameter to ensure positive semi-definite Hessian.", MS_NOT_AVAILABLE);
+	SG_ADD(&m_epsilon, "m_epsilon", "Stopping criteria.", MS_NOT_AVAILABLE);
+}
+
+void CSigmoidCalibration::set_maxiter(index_t maxiter)
+{
+	m_maxiter = maxiter;
+}
+
+index_t CSigmoidCalibration::get_maxiter()
+{
+	return m_maxiter;
+}
+
+void CSigmoidCalibration::set_minstep(float64_t minstep)
+{
+	m_minstep = minstep;
+}
+
+float64_t CSigmoidCalibration::get_minstep()
+{
+	return m_minstep;
+}
+
+void CSigmoidCalibration::set_sigma(float64_t sigma)
+{
+	m_sigma = sigma;
+}
+
+float64_t CSigmoidCalibration::get_sigma()
+{
+	return m_sigma;
+}
+
+void CSigmoidCalibration::set_epsilon(float64_t epsilon)
+{
+	m_epsilon = epsilon;
+}
+
+float64_t CSigmoidCalibration::get_epsilon()
+{
+	return m_epsilon;
+}
+
+bool CSigmoidCalibration::fit_binary(CBinaryLabels* predictions, CBinaryLabels* targets)
+{
+	m_sigmoid_as.resize_vector(1);
+	m_sigmoid_bs.resize_vector(1);
+	auto sigmoid_params = CStatistics::fit_sigmoid(predictions->get_values(), targets->get_labels(), 
+		m_maxiter, m_minstep, m_sigma, m_epsilon);
+
+	m_sigmoid_as[0] = sigmoid_params.a;
+	m_sigmoid_bs[0] = sigmoid_params.b;
+
+	return true;
+}
+
+CBinaryLabels*
+CSigmoidCalibration::calibrate_binary(CBinaryLabels* predictions)
+{
+	CStatistics::SigmoidParamters params; 
+	params.a = m_sigmoid_as[0];
+	params.b = m_sigmoid_bs[0];
+
+	/** Convert predictions to probabilties. */
+	auto values = calibrate_values(predictions->get_values(), params);
+	
+	CBinaryLabels* calibrated_predictions = new CBinaryLabels(values);
+
+	return calibrated_predictions;
+}
+
+bool CSigmoidCalibration::fit_multiclass(CMulticlassLabels* predictions, CMulticlassLabels* targets)
+{
+	index_t num_classes =
+	    predictions->get_num_classes();
+
+	m_sigmoid_as.resize_vector(num_classes);
+	m_sigmoid_bs.resize_vector(num_classes);
+
+	 /** Fit and store parameters for for each class seperately. */
+
+	for (index_t i = 0; i < num_classes; ++i)
+	{
+		auto class_targets = targets->get_binary_for_class(i);
+		auto pred_values = predictions->get_confidences_for_class(i);
+		auto target_labels = class_targets->get_labels();
+
+
+		auto sigmoid_params = CStatistics::fit_sigmoid(pred_values, target_labels, 
+			m_maxiter, m_minstep, m_sigma, m_epsilon);
+		m_sigmoid_as[i] = sigmoid_params.a;
+		m_sigmoid_bs[i] = sigmoid_params.b;
+
+		SG_UNREF(class_targets);
+	}
+
+	return true;
+}
+
+CMulticlassLabels*
+CSigmoidCalibration::calibrate_multiclass(CMulticlassLabels* predictions)
+{
+	index_t num_classes = predictions->get_num_classes();
+	index_t num_samples = predictions->get_num_labels();
+
+	/** Matrix to temporarily store the probabilities. */
```

Comment:
I would never write the type of a variable in a comment, but minor

Reply:
Yeah, it seems does seem very redundant. Will change this.